src/char/int_storage.c

```diff
@@ -174,15 +304,37 @@ int inter_storage_guild_storage_fromsql(int guild_id, struct guild_storage* p)
 	return 0;
 }
 
+/**
+ * Ensures storage data entity for a character.
+ * @see DBCreateData
+ */
+static struct DBData inter_storage_ensure_account_storage(union DBKey key, va_list args)
+{
+	struct storage_data *stor = NULL;
+
+	CREATE(stor, struct storage_data, 1);
+
+	stor->save = false;
+	stor->aggregate = 0;
+
+	VECTOR_INIT(stor->item);
+
+	return DB->ptr2data(stor);
+}
+
 //---------------------------------------------------------
 // storage data initialize
 int inter_storage_sql_init(void)
 {
+	inter_storage->account_storage = idb_alloc(DB_OPT_RELEASE_DATA);
+
 	return 1;
 }
 // storage data finalize
 void inter_storage_sql_final(void)
 {
+	inter_storage->account_storage->destroy(inter_storage->account_storage, NULL);
```

Comment:
This should call a destructor function that calls VECTOR_CLEAR() on each remaining entry before it gets autoreleased